test/parallel_api/numeric/numeric.ops/exclusive_scan_by_segment.pass.cpp

```diff
@@ -168,25 +171,29 @@ DEFINE_TEST(test_exclusive_scan_by_segment)
 
         // call algorithm with no optional arguments
         initialize_data(keys_first, vals_first, val_res_first, n);
-        auto res1 = oneapi::dpl::exclusive_scan_by_segment(exec, keys_first, keys_last, vals_first, val_res_first);
+        [[maybe_unused]] auto res1 =
+            oneapi::dpl::exclusive_scan_by_segment(exec, keys_first, keys_last, vals_first, val_res_first);
         check_values(keys_first, val_res_first, zero, n);
 
         // call algorithm with initial value
         initialize_data(keys_first, vals_first, val_res_first, n);
-        auto res2 = oneapi::dpl::exclusive_scan_by_segment(exec, keys_first, keys_last, vals_first, val_res_first, init);
+        [[maybe_unused]] auto res2 =
+            oneapi::dpl::exclusive_scan_by_segment(exec, keys_first, keys_last, vals_first, val_res_first, init);
         check_values(keys_first, val_res_first, init, n);
 
         // call algorithm with equality comparator
         initialize_data(keys_first, vals_first, val_res_first, n);
-        auto res3 = oneapi::dpl::exclusive_scan_by_segment(exec, keys_first, keys_last, vals_first, val_res_first, zero,
-                                                           [](KeyT first, KeyT second) { return first == second; });
+        [[maybe_unused]] auto res3 =
+            oneapi::dpl::exclusive_scan_by_segment(exec, keys_first, keys_last, vals_first, val_res_first, zero,
+                                                   [](KeyT first, KeyT second) { return first == second; });
         check_values(keys_first, val_res_first, zero, n);
 
         // call algorithm with addition operator
         initialize_data(keys_first, vals_first, val_res_first, n);
-        auto res4 = oneapi::dpl::exclusive_scan_by_segment(exec, keys_first, keys_last, vals_first, val_res_first,
-                                                           init, [](KeyT first, KeyT second) { return first == second; },
-                                                           [](ValT first, ValT second) { return first + second; });
+        [[maybe_unused]] auto res4 = oneapi::dpl::exclusive_scan_by_segment(
```

Comment:
This is a slightly different, but again it's not maybe_unused. It's unused. Please investigate all similar cases but I believe we can just drop res4, res3 and all other variables like that in this file or in others (or probably it's good indication that we don't check the returned result anyhow, and probably we should)

Reply:
Reverted to initial state - will fix these warnings later.